rdoxygen-design.md

```diff
@@ -0,0 +1,126 @@
+# Doxygen Documentation of C++ Libraries in R
+
+## Design Considerations
+
+- *Features.*  
+
+    * **rdoxygen** allows R package developers to add [Doxygen](http://www.stack.nl/~dimitri/doxygen/index.html) documentation to their package for C++ source code (typically found in `src` or `inst/include`).  
+    
+    * Optionally (*should it be optional?*), this documentation can be accessed by users of the package via R [vignettes](http://r-pkgs.had.co.nz/vignettes.html).
+    
+    * Should the package allow more than one Doxygen documentation per package?  It seems like the need for this is rather unlikely.  If it is decied that **rdoxygen** does not support this feature, then its interface could be somewhat simplified (see below).
```

Comment:
Very interesting question. Theoretically it should be possible to implement one module of your package in Fortran and another in C. In this case it would be create to also have separate doxygen documentation. But: How often does this happen and is it really necessary to complicate rdoxygen with this feature?

Reply:
The more I think about it, the more I like it: Maybe you want to have to different kinds of doxygen output for the same code?

Reply:
Agreed: it would be annoying to want to have two Doxy-docs in the package but not be able to do so with **rdoxygen**.  Besides, the `doxy_` functions as currently suggested are set up to make single Doxy-doc completely straightfoward; multiple Doxy-docs require more inputs, but are entirely feasible.